lib/src/meta.dart

```diff
@@ -25,104 +31,117 @@ typedef GeomEachCallback = dynamic Function(
+    currentGeometry = _getGeometry(geometryObject, geometryIndex);
+    runGeomEachCallbacks(currentGeometry, callback, featureIndex,
+        featureProperties, featureBBox, featureId);
+  }
+}
 
-    for (var g = 0; g < stopG; g++) {
-      geometry = isGeometryCollection
-          ? geometryMaybeCollection.geometries[g]
-          : geometryMaybeCollection;
-      if (geometry == null) {
+void runGeomEachCallbacks(
+    GeometryType<dynamic> currentGeometry,
+    GeomEachCallback callback,
+    int featureIndex,
+    Map<String, dynamic> featureProperties,
+    BBox? featureBBox,
+    dynamic featureId) {
+  switch (currentGeometry.type) {
+    case GeoJSONObjectTypes.point:
+    case GeoJSONObjectTypes.lineString:
+    case GeoJSONObjectTypes.multiPoint:
+    case GeoJSONObjectTypes.polygon:
+    case GeoJSONObjectTypes.multiLineString:
+    case GeoJSONObjectTypes.multiPolygon:
+      if (callback(
+            currentGeometry,
+            featureIndex,
+            featureProperties,
+            featureBBox,
+            featureId,
+          ) ==
+          false) {
+        throw ShortCircuit();
+      }
+      break;
+    case GeoJSONObjectTypes.geometryCollection:
+      for (var j = 0;
+          j < (currentGeometry as GeometryCollection).geometries.length;
+          j++) {
         if (callback(
-              null,
+              (currentGeometry as GeometryCollection).geometries[j],
               featureIndex,
               featureProperties,
               featureBBox,
               featureId,
             ) ==
             false) {
-          return;
+          throw ShortCircuit();
         }
-        continue;
       }
-      switch (geometry.type) {
-        case GeoJSONObjectTypes.point:
-        case GeoJSONObjectTypes.lineString:
-        case GeoJSONObjectTypes.multiPoint:
-        case GeoJSONObjectTypes.polygon:
-        case GeoJSONObjectTypes.multiLineString:
-        case GeoJSONObjectTypes.multiPolygon:
-          if (callback(
-                geometry,
-                featureIndex,
-                featureProperties,
-                featureBBox,
-                featureId,
-              ) ==
-              false) {
-            return;
-          }
-          break;
-        case GeoJSONObjectTypes.geometryCollection:
-          for (var j = 0; j < geometry.geometries.length; j++) {
-            if (callback(
-                  geometry.geometries[j],
-                  featureIndex,
-                  featureProperties,
-                  featureBBox,
-                  featureId,
-                ) ==
-                false) {
-              return;
-            }
-          }
-          break;
-        default:
-          throw ('Unknown Geometry Type');
-      }
-    }
-    // Only increase `featureIndex` per each feature
-    featureIndex++;
+      break;
+    default:
+      throw Exception('Unknown Geometry Type');
   }
 }
+
+int _getFeaturesLength(FeatureCollection featureCollection) {
```

Comment:
I think this should be inlined up above instead now that it's just the featureCollection.features.length

Reply:
fixed